x/posts/internal/types/msgs.go

```diff
@@ -173,6 +187,57 @@ func (msg MsgEditPost) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Editor}
 }
 
+// ----------------------
+// --- MsgAnswerPoll
+// ----------------------
+
+// MsgAnswerPoll defines the AnswerPoll message
+type MsgAnswerPoll struct {
+	PostID      PostID         `json:"post_id"`
+	UserAnswers []uint         `json:"provided_answers"`
```

Comment:
I would rename this to be `answers` instead of `provided_answers` as the latter may be confused with the `provided_answer` field inside the `Post` object.

```suggestion
	UserAnswers []uint         `json:"answers"`
```